### PR TITLE
Waste Disagg Phase 3: Implementing disagg for IO tables

### PR DESCRIPTION
**Pull request by jvendries**
cc: https://github.com/orgs/cornerstone-data/projects/18/views/1
Closes: https://github.com/cornerstone-data/bedrock/issues/251

## What changed? Why?

Added comprehensive waste disaggregation functionality for EEIO matrices with full test coverage. This implements the core disaggregation logic that splits the aggregate waste sector (562000) into detailed subsectors across Make, Use, Value Added, and Final Demand matrices.

**New functionality:**
- `apply_waste_disagg_to_V()` - Disaggregates Make matrix using intersection, column, and row weights
- `apply_waste_disagg_to_U()` - Disaggregates Use matrices (domestic and import) with proper handling of VA/FD exclusions  
- `apply_waste_disagg_to_VA()` - Splits Value Added rows across waste industry subsectors
- `apply_waste_disagg_to_Ytot()` - Disaggregates Final Demand with FD-specific weights and fallback logic

**Key features:**
- Handles cases where matrices are already disaggregated by re-aggregating first
- Preserves mass balance across all transformations
- Uses specific weights when available, falls back to uniform distribution otherwise
- Excludes VA rows from Use column disaggregation and FD columns from Use row disaggregation

## Testing

- **Unit tests** with synthetic data verify core disaggregation logic, mass preservation, and edge cases
- **Integration tests** using real 2017 CSV weights validate against actual BEA data structures
- **Pytest markers** separate fast unit tests from slower integration tests that require data files
- All tests confirm mass balance preservation and proper application of weight matrices

### CODE REVIEW

**Comment by jvendries** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/cornerstone-data/bedrock/250?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#250** <a href="https://app.graphite.com/github/pr/cornerstone-data/bedrock/250?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/cornerstone-data/bedrock/250?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#243** <a href="https://app.graphite.com/github/pr/cornerstone-data/bedrock/243?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#237** <a href="https://app.graphite.com/github/pr/cornerstone-data/bedrock/237?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: bedrock/transform/eeio/waste_disaggregation.py [hunk 1]
```diff
@@ -0,0 +1,477 @@
...
+
+from typing import cast
+
+import numpy as np
+import pandas as pd
+
+from bedrock.extract.disaggregation.waste_weights import WasteDisaggWeights
+
+
+def _assert_non_waste_unchanged(
```
Comment (bl-young) [inline bedrock/transform/eeio/waste_disaggregation.py]: thanks for adding this
